core/src/main/java/me/leoko/advancedban/utils/DynamicDataSource.java

```diff
@@ -18,13 +19,21 @@ public DynamicDataSource(boolean preferMySQL) {
             String properties = mi.getString(mi.getMySQLFile(), "MySQL.Properties", "verifyServerCertificate=false&useSSL=false&useUnicode=true&characterEncoding=UTF-8");
             int port = mi.getInteger(mi.getMySQLFile(), "MySQL.Port", 3306);
 
+            if(Universal.get().getServerType() == ServerType.VELOCITY) {
+                config.setDriverClassName("org.mariadb.jdbc.Driver");
+            }
+
             config.setJdbcUrl("jdbc:mysql://" + ip + ":" + port + "/" + dbName + "?"+properties);
             config.setUsername(usrName);
             config.setPassword(password);
         } else {
             config.setJdbcUrl("jdbc:hsqldb:file:" + mi.getDataFolder().getPath() + "/data/storage;hsqldb.lock_file=false");
             config.setUsername("SA");
             config.setPassword("");
+
+            if(Universal.get().getServerType() == ServerType.VELOCITY) {
+                config.setDriverClassName("org.hsqldb.jdbc.JDBCDriver");
```

Comment:
Good catch with setting the driver class name. For HSQLDB, setting the driver class name should actually be unconditional and applied for all platforms. (For the MariaDB driver, the code you have is correct because AdvancedBan uses the inbuilt driver on platforms besides Velocity)

Reply:
> For the commands, I believe the same limitations are present on BungeeCord. The UUIDManager will be used to lookup the target player's UUID depending on the FetcherMode, which is by default detected based on the environment. If you are testing your PR in offline mode, mi.isOnlineMode() will be false and no UUIDs will be looked up.
> 
> I realized you are right with regards to setting the JDBC driver class name - it ensures AdvancedBan uses its own relocated driver, even when another plugin shades a different version of the same driver.
> 
> There are a couple other comments that still need to be addressed, but otherwise this PR is close to ready. We can then get one more reviewer (probably Leoko)

This is the issue - I'm testing in online mode. I'll have a look into the UUIDManager and see if I can figure it out.

Realised I missed a few of the requested changes above, so I'll get them done for the next commit.

Reply:
Issue is Velocity has no built-in method of fetching a UUID - Will write something using the Mojang API in the next little bit in order to fetch a uuid.

Reply:
That's exactly what UUIDManager does - makes a request to the Mojang API or another web service as configured

Reply:
Well, need to diagnose why it's not working then, xD